### ISSUE TITLE
node defaults when targetting node

### ISSUE THREAD

**Issue by travi** [webpack/webpack#7553]
## Feature request

**What is the expected behavior?**
When `target` is set to `'node'`, I expected `__dirname` to behave as normal since the code would be running in a node environment. Instead, it is ["mock"ed](https://github.com/webpack/docs/wiki/configuration#node) to resolve to `/` by default. I can see how this would make sense if targeting a browser, but was surprising when targeting node.

**What is motivation or use case for adding/changing the behavior?**
I think defaulting to normal node behavior is a less surprising default when targeting node.

**How should this be implemented in your opinion?**
Since the issue I ran into specifically was related to `__dirname`, I haven't given enough thought to all of the [node](https://webpack.js.org/configuration/node/#node) options, but since it is mention that 

> Since webpack 3.0.0, the node option may be set to false to completely turn off the NodeStuffPlugin and NodeSourcePlugin plugins.

I could see setting `node` to `false` by default when `target` is set to `node`, but I'm open to feedback about where that isn't appropriate.

**Are you willing to work on this yourself?**
yes with guidance, but this would be fairly low priority for me since i was able to configure it to work for our project. mostly wanted to bring some attention to it in hopes of saving someone else the time I had to invest to get that working for us.

**Comment by chrisdothtml** [comment]
> defaulting to normal node behavior

By this do you mean

1. letting node to provide the value and webpack just not intervening at all, or
2. providing the path to the source file

Option 1 can lead to confusing source code, for example, if your source file `src/deep/dir/module.js` looks like this:

```js
const templatePath = path.join(__dirname, 'template.txt')
```

but the value of `__dirname` actually resolves to `dist` at runtime. It's fair to say that anyone reading that source code, at first glance, would expect the value to be `src/deep/dir`

**Comment by travi** [comment]
sounds like i wasn't clear enough with the suggestion. the specific adjustment that i made to get the behavior that i expected was as follows:

```
    target: 'node',
    node: {__dirname: false},
```

i simply expected that to be the default behavior, w/o the need to configure it specifically. the webpack behavior, once configured is already as expected.

**Comment by sokra** [comment]
this was fixed in #9172
